### PR TITLE
Building controller modules (#1273)

### PR DESCRIPTION
**Pull request by EMcWhinn**
* Building controller modules

* Further work on Chapter 2 and 3 for controller

* Adding chapters 4-9

### CODE REVIEW

File: downstream/modules/eda/proc-eda-view-activation-output.adoc
Comment (EMcWhinn) [inline downstream/modules/eda/proc-eda-view-activation-output.adoc:1]: Ignore this file.

File: downstream/assemblies/platform/assembly-controller-getting-started-credentials.adoc [hunk 1]
```diff
@@ -0,0 +1,9 @@
+[id="controller-getting-started-credentials"]
```
Comment (ianf77) [inline downstream/assemblies/platform/assembly-controller-getting-started-credentials.adoc:1]: Can you remove the getting-started(s), identifying the document doesn't make for easy reuse.

File: downstream/modules/eda/proc-eda-view-activation-output.adoc [hunk 2]
```diff
@@ -5,7 +5,7 @@
 You can view the output of the activations in the *History* tab.
 
 .Procedure
-. Select the *History* tab to access the list of all the activation instances.
+. Select the btn:[History] tab to access the list of all the activation instances.
```
Comment (ianf77) [inline downstream/modules/eda/proc-eda-view-activation-output.adoc:8]: I don't think we use the btn:[] approach for tabs. i think the original is correct.

File: downstream/modules/platform/proc-controller-getting-started-edit-project.adoc [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+.Procedure
+. Open the project to edit it by using one of these methods:
+** Go to the *Details* tab. 
+Click btn:[Edit].
+** Go to the *Projects* list view. 
+Click btn:[Edit} next to the project name and edit the appropriate details.
+. Save your changes
+
+
+If you want to fetch the project for the latest changes, you can manually start an SCM sync for this project:
```
Comment (ianf77) [inline downstream/modules/platform/proc-controller-getting-started-edit-project.adoc:17]: Have you said somewhere what an SCM sync is?

File: downstream/modules/platform/proc-controller-getting-started-inventory.adoc [hunk 4]
```diff
@@ -0,0 +1,21 @@
+[id="controller-getting-started-inventory"]
+
+= Creating a new Inventory
+
+The Inventories window displays a list of the inventories that are currently available. 
+You can sort the inventory list by name and searched type, organization, description, owners and modifiers of the inventory, or additional criteria.
+
+.Procedure
+. To view existing inventories, click btn:[Inventories] in the navigation bar.
```
Comment (ianf77) [inline downstream/modules/platform/proc-controller-getting-started-inventory.adoc:9]: I'm not sure a) whether they're buttons to click, or menu items to select. Or whether it's a navigation bar or the navigation menu. Perhaps we could check with Donna on the accessibility implications?

File: downstream/modules/platform/proc-controller-importing-subscriptions.adoc [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+
+.Procedure
+
+. Launch controller for the first time. 
+The *Subscription Management* screen is displayed.
++
+image::controller-gs-subscription-management.png[Subscription Management]
++
+. Retrieve and and import your subscription by completing either of the following steps:
+.. If you have link:https://access.redhat.com/management/subscription_allocations[obtained a subscription manifest], upload it by browsing to the location where the file is saved (the subscription manifest is the complete .zip file, and not only its component parts).
```
Comment (ianf77) [inline downstream/modules/platform/proc-controller-importing-subscriptions.adoc:19]: upload it by browsing to, or by navigating to?

File: downstream/modules/platform/proc-controller-importing-subscriptions.adoc [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
++
+[NOTE]
+====
+If the *Browse* option in the subscription manifest option is disabled, clear the username and password fields to enable it.
+====
++
+The subscription metadata is then retrieved from the RHSM/Satellite API, or from the manifest provided. 
+If multiple subscription counts were applied in a single installation, the controller combines the counts but uses the earliest expiration date as the expiry (at which point you must refresh your subscription).
```
Comment (ianf77) [inline downstream/modules/platform/proc-controller-importing-subscriptions.adoc:27]: installation, {ControllerName} combines

File: downstream/modules/platform/proc-controller-importing-subscriptions.adoc [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+.. If you are using your Red Hat customer credentials, enter your username and password on the license page. 
+Use your Satellite username or password if your automation controller cluster nodes are registered to Satellite with Subscription Manager. 
+After you enter your credentials, click *Get Subscriptions*.
++
+{ControllerNameStart} retrieves your configured subscription service. 
+Then, it prompts you to choose the subscription that you want to run and entitles the controller with that metadata.
```
Comment (ianf77) [inline downstream/modules/platform/proc-controller-importing-subscriptions.adoc:33]: and entitles -> and applies that metadata to {ControllerName}.

File: downstream/modules/platform/proc-controller-importing-subscriptions.adoc [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+[NOTE]
+====
+You can change your analytics data collection preferences at any time, as described in the link:https://docs.ansible.com/automation-controller/4.4/html/administration/usability_data_collection.html#usability-data-collection[Usability Analytics and Data Collection] section.
+====
++
+. After you have specified your tracking and Insights preferences, click btn:[Next] to proceed to the End User Agreement.
+. Review and check the *I agree to the End User License Agreement* checkbox and click btn:[Submit].
+
+After your subscription is accepted, {ControllerName} displays the subscription details and opens the Dashboard. 
+To return to the Subscription settings screen from the Dashboard, click btn:[Settings] on the navigation menu and select btn:[Subscription settings] from the btn:[Subscription] option.
```
Comment (ianf77) [inline downstream/modules/platform/proc-controller-importing-subscriptions.adoc:55]: again, navigation menu or navigation bar?

File: downstream/modules/platform/proc-controller-importing-subscriptions.adoc [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+
+image::controller-gs-licenseaccepted.png[Subscription Details]
+
+.Troubleshooting
+
+When your subscription expires (you can check this in the Subscription details of the Subscription settings window), you must renew it in {ControllerName} by one of the preceding two methods.
+
+If you encounter the “Error fetching licenses'” message, you need the proper permissions required for the Satellite user.
```
Comment (ianf77) [inline downstream/modules/platform/proc-controller-importing-subscriptions.adoc:63]: you must have the proper permissions

File: downstream/modules/platform/proc-controller-importing-subscriptions.adoc [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+image::controller-gs-licenseaccepted.png[Subscription Details]
+
+.Troubleshooting
+
+When your subscription expires (you can check this in the Subscription details of the Subscription settings window), you must renew it in {ControllerName} by one of the preceding two methods.
+
+If you encounter the “Error fetching licenses'” message, you need the proper permissions required for the Satellite user. 
+The {ControllerName} administrator requires this to apply a subscription.
+
+The Satellite username or password is used to query the Satellite API for existing subscriptions.
```
Comment (ianf77) [inline downstream/modules/platform/proc-controller-importing-subscriptions.adoc:66]: Either the Satellite.... one or both?

File: downstream/modules/platform/proc-controller-obtaining-subscriptions.adoc [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+
+If you already have a subscription to a Red Hat product, you can acquire an automation controller subscription through that subscription. 
+If not, you can request a link:https://www.redhat.com/en/technologies/management/ansible/try-it[trial subscription].  
+
+.Procedure
+
+* If you already have a {PlatformName} subscription, use your Red Hat customer credentials when you launch the controller to access your subscription information. 
+See xref:controller-importing-subscriptions[Importing a subscription].
+
+* If you have a non-Ansible Red Hat or Satellite subscription, access the automation controller with one of these methods:
```
Comment (ianf77) [inline downstream/modules/platform/proc-controller-obtaining-subscriptions.adoc:13]: access {ControllerName} with

File: downstream/modules/platform/proc-controller-obtaining-subscriptions.adoc [hunk 12]
```diff
@@ -0,0 +1,22 @@
...
+** Enter your username and password on the license page.
+** Obtain a subscriptions manifest from the link:https://access.redhat.com/management/subscription_allocations[Subscription Allocations] page on the Red Hat Customer Portal. 
+For detailed information, see link:http://docs.ansible.com/automation-controller/4.4/html/userguide/import_license.html#obtain-sub-manifest[Obtaining a subscriptions manifest] in the Automation Controller User Guide.
+** If you do not have a {PlatformName} subscription, go to link:https://www.redhat.com/en/technologies/management/ansible/trial[Try Red Hat Ansible Automation Platform] and request a trial subscription.
+
+.Additional resources
+
+To understand what is supported with your subscription, see link:http://docs.ansible.com/automation-controller/4.4/html/userguide/license-support.html#licenses-feat-support[Red Hat Ansible Automation Platform Controller Licensing, Updates, and Support].
```
Comment (ianf77) [inline downstream/modules/platform/proc-controller-obtaining-subscriptions.adoc:21]: There's an xref for this now. But probably better to wait until the UG is complete.

File: downstream/modules/platform/proc-controller-getting-started-edit-project.adoc (hunk 3 above)
Comment (EMcWhinn) [reply downstream/modules/platform/proc-controller-getting-started-edit-project.adoc:17]: This is defined as "source code management" in the `controller-projects` assembly.

File: downstream/modules/platform/proc-controller-edit-job-template.adoc [hunk 13]
```diff
@@ -0,0 +1,17 @@
+[id="controller-edit-job-templates"]
+
+= Editing a job template
+
+As part of the initial setup, you can leave the default *Demo Job Template* as it is, but you can edit it later.
+
+.Procedure
+
+. Open the template to edit it by using one of these methods:
+** Go to the menu:Details[Edit].
```
Comment (ariordan-redhat) [inline downstream/modules/platform/proc-controller-edit-job-template.adoc:10]:
```suggestion
** Click menu:Details[Edit].
```

File: downstream/modules/platform/proc-controller-importing-subscriptions.adoc [hunk 14]
```diff
@@ -0,0 +1,84 @@
...
+[NOTE]
+====
+You can change your analytics data collection preferences at any time, as described in the link:https://docs.ansible.com/automation-controller/4.4/html/administration/usability_data_collection.html#usability-data-collection[Usability Analytics and Data Collection] section.
+====
++
+. After you have specified your tracking and Insights preferences, click btn:[Next] to proceed to the End User Agreement.
+. Review and check the *I agree to the End User License Agreement* checkbox and click btn:[Submit].
+
+After your subscription is accepted, {ControllerName} displays the subscription details and opens the Dashboard. 
+To return to the Subscription settings screen from the Dashboard, from the navigation panel select menu:Settings[Subscription settings] from the btn:[Subscription] option.
```
Comment (ariordan-redhat) [inline downstream/modules/platform/proc-controller-importing-subscriptions.adoc:55]:
```suggestion
To return to the Subscription settings screen from the Dashboard, select menu:Settings[Subscription settings] from the btn:[Subscription] option in the navigation panel.
```

File: downstream/modules/platform/proc-controller-importing-subscriptions.adoc [hunk 15]
```diff
@@ -0,0 +1,84 @@
...
+
+image::controller-gs-licenseaccepted.png[Subscription Details]
+
+.Troubleshooting
+
+When your subscription expires (you can check this in the Subscription details of the Subscription settings window), you must renew it in {ControllerName} by one of the preceding two methods.
+
+If you encounter the "Error fetching licenses" message, you must have the proper permissions required for the Satellite user.
```
Comment (ariordan-redhat) [inline downstream/modules/platform/proc-controller-importing-subscriptions.adoc:63]:
```suggestion
If you encounter the "Error fetching licenses" message, check that you have the proper permissions required for the Satellite user. 
```

File: downstream/modules/platform/proc-controller-logging-in.adoc [hunk 16]
```diff
@@ -0,0 +1,12 @@
+[id="controller-logging-in"]
+
+.Procedure
+
+. With the login information provided after your installation completed, open a web browser and log in to the {ControllerName} by navigating to its server URL at: https://<CONTROLLER_SERVER_NAME>/
```
Comment (ariordan-redhat) [inline downstream/modules/platform/proc-controller-logging-in.adoc:5]:
```suggestion
. With the login information provided after your installation completed, open a web browser and log in to the {ControllerName} by navigating to its server URL at: \https://<CONTROLLER_SERVER_NAME>/

Does the link need to be live?  Putting a backslash in front of the URL prevents automatic linking
```

File: downstream/modules/platform/proc-controller-obtaining-subscriptions.adoc [hunk 17]
```diff
@@ -0,0 +1,22 @@
+[id="controller-obtaining-subscriptions"]
+
+= Obtaining an authorized Ansible {ControllerName} subscription
```
Comment (ariordan-redhat) [inline downstream/modules/platform/proc-controller-obtaining-subscriptions.adoc:3]: Can customers get a controller subscription, or do they need an AAP subscription?

File: downstream/modules/platform/proc-controller-obtaining-subscriptions.adoc [hunk 18]
```diff
@@ -0,0 +1,22 @@
...
+
+If you already have a subscription to a Red Hat product, you can acquire an {ControllerName} subscription through that subscription. 
+If not, you can request a link:https://www.redhat.com/en/technologies/management/ansible/try-it[trial subscription].  
+
+.Procedure
+
+* If you already have a {PlatformName} subscription, use your Red Hat customer credentials when you launch the controller to access your subscription information.
```
Comment (ariordan-redhat) [inline downstream/modules/platform/proc-controller-obtaining-subscriptions.adoc:10]:
```suggestion
* If you already have a {PlatformName} subscription, use your Red Hat customer credentials when you launch {ControllerName} to access your subscription information. 
```

File: downstream/modules/platform/proc-controller-run-job-template.adoc [hunk 19]
```diff
@@ -0,0 +1,28 @@
...
+The initial job launch generates a status page, which updates automatically using {ControllerName}'s Live Event feature, until the job is complete. 
+
+For more information on the job results, see link:http://docs.ansible.com/automation-controller/4.4/html/userguide/jobs.html#ug-jobs[Jobs] in the Automation Controller User Guide.
+
+.Additional resources
+
+To learn more about these {ControllerName} features or to learn about administration tasks and the controller API, see the following documentation sets:
+
+* link:http://docs.ansible.com/automation-controller/4.4/html/userguide/index.html#ug-star[Automation Controller User Guide]
+* link:https://access.redhat.com/documentation/en-us/red_hat_ansible_automation_platform/2.0-ea/html-single/red_hat_ansible_automation_platform_installation_guide/index[Red Hat Ansible Automation Platform Installation Guide]
```
Comment (ariordan-redhat) [inline downstream/modules/platform/proc-controller-run-job-template.adoc:24]:
```suggestion
* link:https://access.redhat.com/documentation/en-us/red_hat_ansible_automation_platform/{PlatformVers}/html-single/red_hat_ansible_automation_platform_installation_guide/index[Red Hat Ansible Automation Platform Installation Guide]
```

File: downstream/modules/platform/proc-controller-run-job-template.adoc [hunk 20]
```diff
@@ -0,0 +1,28 @@
...
+
+The initial job launch generates a status page, which updates automatically using {ControllerName}'s Live Event feature, until the job is complete. 
+
+For more information on the job results, see link:http://docs.ansible.com/automation-controller/4.4/html/userguide/jobs.html#ug-jobs[Jobs] in the Automation Controller User Guide.
+
+.Additional resources
+
+To learn more about these {ControllerName} features or to learn about administration tasks and the controller API, see the following documentation sets:
+
+* link:http://docs.ansible.com/automation-controller/4.4/html/userguide/index.html#ug-star[Automation Controller User Guide]
```
Comment (ariordan-redhat) [inline downstream/modules/platform/proc-controller-run-job-template.adoc:23]:
```suggestion
* link:http://docs.ansible.com/automation-controller/latest/html/userguide/index.html#ug-star[Automation Controller User Guide]
```

Optional: these links will probably be replaced once the migration to asciidoctor is done, and  4.4 will always be the most recent version on docs.ansible.com

File: downstream/modules/platform/proc-controller-sync-project.adoc [hunk 21]
```diff
@@ -0,0 +1,18 @@
+[id="controller-sync-project"]
+
+= Syncing a project
+
+If you want to fetch the project for the latest changes, you can manually start an SCM sync for this project.
```
Comment (ariordan-redhat) [inline downstream/modules/platform/proc-controller-sync-project.adoc:5]:
```suggestion
If you want to fetch the latest changes in a project, you can manually start an SCM sync for this project.
```

File: downstream/modules/platform/proc-controller-viewing-dashboard.adoc [hunk 22]
```diff
@@ -0,0 +1,30 @@
+[id="controller-viewing-dashboard"]
+
+= Viewing the Dashboard
+
+.Procedure
+
+* Click the Menu icon to hide or display the navigation panel.
```
Comment (ariordan-redhat) [inline downstream/modules/platform/proc-controller-viewing-dashboard.adoc:7]:
```suggestion
* Click the *Menu* icon to hide or display the navigation panel.
```

File: downstream/modules/platform/proc-controller-viewing-dashboard.adoc [hunk 23]
```diff
@@ -0,0 +1,30 @@
...
+** On the main Dashboard view, a summary appears listing your current *Job status*.
```
Comment (ariordan-redhat) [inline downstream/modules/platform/proc-controller-viewing-dashboard.adoc:8]:
```suggestion
** On the main dashboard, a summary appears, listing your current *Job status*. 
```

File: downstream/modules/platform/proc-controller-viewing-dashboard.adoc [hunk 24]
```diff
@@ -0,0 +1,30 @@
+[id="controller-viewing-dashboard"]
+
+= Viewing the Dashboard
+
+.Procedure
+
+* Click the Menu icon to hide or display the navigation panel.
+** On the main Dashboard view, a summary appears listing your current *Job status*. 
+You can filter the job status within a period of time or by job type. 
+** Also available in their respective tabs are summaries of *Recent Jobs* ran and *Recent Templates* used.
```
Comment (ariordan-redhat) [inline downstream/modules/platform/proc-controller-viewing-dashboard.adoc:10]:
```suggestion
** You can view summaries of *Recent Jobs* and *Recent Templates* in their respective tabs. 
```

File: downstream/modules/platform/proc-controller-add-groups-hosts.adoc [hunk 25]
```diff
@@ -0,0 +1,22 @@
+[id="controller-add-groups-and-hosts"]
+
+== Adding new groups and hosts
```
[the remark this answers is not in the document]
Comment (EMcWhinn) [reply downstream/modules/platform/proc-controller-add-groups-hosts.adoc:3]: Done :)

**Review by ariordan-redhat — APPROVED** [review]
LGTM

**Pull request merged**